kraken-build.go

```diff
@@ -114,6 +117,124 @@ func compileTemplates(krakenDir, tmpDir string) (targets []string, e error) {
 	return
 }
 
+// Like CompileTemplates(), but does so for sources that will end up in u-root
+func uCompileTemplates(krakenDir, tmpDir string) (targets []string, e error) {
+	var files []os.FileInfo
+	re, _ := regexp.Compile(".*\\.go\\.tpl$")
+	// build a list of all of the templates
+	files, e = ioutil.ReadDir(filepath.Join(krakenDir, "kraken"))
+	if e != nil {
+		return
+	}
+	for _, f := range files {
+		if f.Mode().IsRegular() {
+			if re.MatchString(f.Name()) { // ends in .go.tpl?
+				if *verbose {
+					log.Printf("executing template: %s", f.Name())
+				}
+				var target string
+				target, e = compileTemplate(filepath.Join(krakenDir, "kraken", f.Name()), tmpDir)
+				if e != nil {
+					return
+				}
+
+				// Avoid import path errors in generated source files
+				e = lib.DeepSearchAndReplace(filepath.Join(tmpDir, target), "hpc/kraken", "u-root/u-root/cmds/exp/kraken")
+				if e != nil {
+					return
+				}
+
+				targets = append(targets, target)
+			}
+		}
+	}
+	return
+}
+
+// Generate kraken source tree for u-root command from krakenDir into outDir
+func uKraken(outDir, krakenDir string) (targets []string, e error) {
```

Comment:
Can we split this unto sub-functions?

Reply:
The complexity did not seem to be with `uKraken()`, but with `main()` after I added the `if` block checking if `-uroot` is set. After moving this block to a separate function `checkUrootOpt()`, the complexity issue goes away.

I did, however, simplify the large `if`-`else` block in `uKraken()` since there was some unnecessary repeated code.